src/main.rs

```diff
@@ -216,93 +224,111 @@ fn complete_command(
-//             fish_options.append('-l ' + fish_escape_single_quote(option[2:]))
-//         elif option.startswith('-') and len(option) == 2:
-//             # New style short option (-r)
-//             fish_options.append('-s ' + fish_escape_single_quote(option[1:]))
-//         elif option.startswith('-') and len(option) > 2:
-//             # Old style long option (-recursive)
-//             fish_options.append('-o ' + fish_escape_single_quote(option[1:]))
-//
-//     # Determine which options are new (not already in existing_options)
-//     # Then add those to the existing options
-//     existing_options = already_output_completions.setdefault(CMDNAME, set())
-//     fish_options = [opt for opt in fish_options if opt not in existing_options]
-//     existing_options.update(fish_options)
-//
-//     # Maybe it's all for naught
-//     if not fish_options: return
-//
-//     # Here's what we'll use to truncate if necessary
-//     max_description_width = 78
-//     if IS_PY3:
-//         truncation_suffix = '…'
-//     else:
-//         ELLIPSIS_CODE_POINT = 0x2026
-//         truncation_suffix = unichr(ELLIPSIS_CODE_POINT)
-//
-//     # Try to include as many whole sentences as will fit
-//     # Clean up some probably bogus escapes in the process
-//     clean_desc = description.replace("\\'", "'").replace("\\.", ".")
-//     sentences = clean_desc.split('.')
-//
-//     # Clean up "sentences" that are just whitespace
-//     # But don't let it be empty
-//     sentences = [x for x in sentences if x.strip()]
-//     if not sentences: sentences = ['']
-//
-//     udot = lossy_unicode('.')
-//     uspace = lossy_unicode(' ')
-//
-//     truncated_description = lossy_unicode(sentences[0]) + udot
-//     for line in sentences[1:]:
-//         if not line: continue
-//         proposed_description = lossy_unicode(truncated_description) + uspace + lossy_unicode(line) + udot
-//         if len(proposed_description) <= max_description_width:
-//             # It fits
-//             truncated_description = proposed_description
-//         else:
-//             # No fit
-//             break
-//
-//     # If the first sentence does not fit, truncate if necessary
-//     if len(truncated_description) > max_description_width:
-//         prefix_len = max_description_width - len(truncation_suffix)
-//         truncated_description = truncated_description[:prefix_len] + truncation_suffix
-//
-//     # Escape some more things
-//     truncated_description = fish_escape_single_quote(truncated_description)
-//     escaped_cmd = fish_escape_single_quote(CMDNAME)
-//
-//     output_complete_command(escaped_cmd, fish_options, truncated_description, built_command_output)
+const MAX_DESCRIPTION_WIDTH: usize = 78;
+const TRUNCATION_SUFFIX: &'static str = "…";
 
-// TODO args / arg types?
-fn built_command() {
-    unimplemented!()
+fn last_char_index(string: &str) -> Option<usize> {
+    string.rfind(|_| true)
```

Comment:
Is there a reason not to use `str::len`?

Reply:
The last byte index might not not be on a UTF-8 character boundary.

Reply:
Ah, good to know.

Reply:
Ended up not needing this. Misinterpreted the original Python's description truncation code.